mimic/model/nova_objects.py

```diff
@@ -931,6 +933,37 @@ def request_action(self, http_action_request, server_id, absolutize_url):
                                          " while it is in task state other than active",
                                          http_action_request))
 
+        elif 'createImage' in action_json:
+            image_name = action_json['createImage'].get('name')
+            server == self.server_by_id(server_id)
+            links = server.links_json(absolutize_url)
+            server_id = server.server_id
+            image_ref = server.image_ref
+            image = image_store.get_image_by_id(image_ref)
+            image_json = regional_image_collection.get_image(http_action_request,
+                                                             image_ref, absolutize_url)
+            image_dict = loads(image_json)
+            flavor_classes = image_dict['image']['metadata']['flavor_classes']
+            os_type = image_dict['image']['metadata']['os_type']
+            os_distro = image_dict['image']['metadata']['org.openstack__1__os_distro']
+            vm_mode = image_dict['image']['metadata']['vm_mode']
+            disk_config = image_dict['image']['metadata']['auto_disk_config']
+            image_id = str(uuid.uuid4())
```

Comment:
`str` is a red flag for python 3 compatibility.  Does this value represent bytes or text?  Since it's going to be used as an image ID, which is textual (and shows up in `JSON`), you should probably be using `six.text_type` here.